src/libraries/System.Reflection.Metadata/tests/Metadata/TypeSystem/TypeDefinitionTests.cs

```diff
@@ -32,5 +32,41 @@ public void ValidateTypeDefinitionIsNestedWindowsProjection()
                 Assert.Equal(typeDef.Attributes.IsNested(), typeDef.IsNested);
             }
         }
+
+        [Fact]
+        public void ValidateDeclaringType()
+        {
+            var reader = MetadataReaderTests.GetMetadataReader(Misc.Members);
+
+            foreach (var typeDefHandle in reader.TypeDefinitions)
+            {
+                var typeDef = reader.GetTypeDefinition(typeDefHandle);
+                foreach (var nestedTypeHandle in typeDef.GetNestedTypes())
```

Comment:
Can we add a sanity check here to make sure that these foreach statement actually loop through at least one? Like:

```cs
Assert.True(typeDef.GetNestedTypes().Length > 0);
```

Reply:
The test assembly actually [dosen't have any nested types](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Reflection.Metadata/tests/Resources/Misc/Members.cs). I tried to add some but after recompiling it [this assert](https://github.com/dotnet/runtime/blob/c587256d18fa01834508b0e5128b5192b48c08dc/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEReaderTests.cs#L244-L249) failed.

Your suggested assert wouldn't currently work, because not all types have all kinds of members each.

Reply:
Added asserts that `GetTypesWithEvents()` and `GetTypesWithProperties()` are not empty, and commented out the asserts on the nested classes.